packages/react/src/components/radio-button/radio-button.tsx

```diff
@@ -0,0 +1,132 @@
+import { ChangeEvent, FunctionComponent } from 'react';
+import styled from 'styled-components';
+import { focus } from '../../utils/css-state';
+import { VisuallyHidden } from '../visually-hidden/visuallyhidden';
+import { useId } from '../../hooks/use-id';
+
+const StyledInput = styled.input<{ disabled?: boolean }>`
+    appearance: none;
+    background-color: ${({ theme, disabled }) => (disabled ? theme.component['radio-button-disabled-background-color'] : theme.component['radio-button-background-color'])};
+    border: 1px solid ${({ theme, disabled }) => (disabled ? theme.component['radio-button-disabled-border-color'] : theme.component['radio-button-border-color'])};
+    border-radius: 50%;
+    color: ${({ theme, disabled }) => (disabled ? theme.component['radio-button-disabled-background-color'] : theme.component['radio-button-background-color'])};
+    display: inline-block;
+    height: var(--size-1x);
+    margin: 0;
+    position: relative;
+    width: var(--size-1x);
+    
+    ${(theme) => focus(theme, { selector: '+' })}
+     
+    &:checked {
+        border: 2px solid ${({ theme, disabled }) => (disabled ? theme.component['radio-button-disabled-border-color'] : theme.component['radio-button-checked-border-color'])};
+        &::after {
+            background-color: ${({ theme, disabled }) => (disabled ? theme.component['radio-button-disabled-border-color'] : theme.component['radio-button-checked-background-color'])};
+            border-radius: 50%;
+            content: '';
+            display: block;
+            height: var(--size-half);
+            left: 50%;
+            position: absolute;
+            top: 50%;
+            transform: translate(-50%, -50%);
+            width: var(--size-half);
+        }
+    }
+        
+    &:hover {
+        border: 1px solid ${({ theme, disabled }) => (disabled ? theme.component['radio-button-disabled-hover-border-color'] : theme.component['radio-button-hover-border-color'])};
+    }
+`;
+
+interface StyledLabelProps {
+    disabled?: boolean;
+    hasLabel: boolean;
+}
+
+const StyledLabel = styled.label<StyledLabelProps>`
+    font-size: 0.875rem;
+    line-height: 1.5rem;
+    margin-left: var(--spacing-1x);
+`;
```

Comment:
Enleve l'interface, ces props ne sont pas utilise

```suggestion
const StyledLabel = styled.label`
    font-size: 0.875rem;
    line-height: 1.5rem;
    margin-left: var(--spacing-1x);
`;
```

Reply:
`disabled` est utilisé.

Reply:
C'est juste qu'il est passe dans le props, mais on ne fait rien avec. C'est mieux de l'enlever pour le moment, et on l'ajouterais a ce moment la.

```
<StyledLabel
    data-testid={`radiobutton-${inputId}_label`}
    htmlFor={inputId}
    className={`${className}_label`}
    disabled={disabled}  // enleve ici aussi
>
```

Reply:
Désolé mais j'te suis pas pantoute. `disabled` est clairement pris en compte dans les components enfants et sert à les désactiver si `true`.

![image](https://github.com/kronostechnologies/design-elements/assets/6412423/f0f83cfa-64a3-4593-afa4-7dbf9c35ae41)

En plus, le Jira demande de prendre en compte le state `disabled`.

Reply:
Oui pour le Component dans l'ensemble, mais pas pour l'element StyledLabel